content/posts/dctf-2021-hotel-rop-ret2libc-pie.md

```diff
@@ -0,0 +1,282 @@
+---
+title: "dCTF 2021 - Hotel ROP"
+date: 2021-05-24T20:22:21-05:00
+draft: false
+subtitle: "Returning 2 LIBC in a PIE enabled binary"
+tags: ["pie", "aslr", "easy", "elf", "libc"]
+author: "<a href='https://caprinux.github.io'>elma</a>"
+---
+Today, we will be looking at a pwn challenge from **dCTF 2021** which features ret2libc exploitation with a little twist of a `PIE-enabled` binary. The following PwnTools features will be introduced here:
+- `pwnlib.rop` to help us craft ROP chains
+- `pwnlib.elf` to make finding addresses quick and easy
+- and many more little modules from `pwntools` to help us pwn faster ~
+
+<br>
+
+## Challenge Description
+---
+> They say programmers' dream is California. And because they need somewhere to stay, we've built a hotel!
+>
+> Attachments: [hotel_rop](/content/files/hotel_rop)
+
+<br>
+
+## Getting Started
+---
+
+For this challenge, we are provided with a binary and nothing else.
+
+We quickly check the security features of the binary with `checksec hotel_rop` which returns

```

Comment:
We should specify `pwn checksec` here since this is how it's installed for most users.  `checksec` refers to `checksec.sh` on some distributions.